cluster/aws/util.sh

```diff
@@ -330,6 +328,27 @@ function detect-trusty-image () {
   fi
 }
 
+# Detects the RootDevice to use in the Block Device Mapping (considering the AMI)
+#
+# Vars set:
+#   MASTER_BLOCK_DEVICE_MAPPINGS
+#   NODE_BLOCK_DEVICE_MAPPINGS
+#
+function detect-root-device {
+  local master_image=${AWS_IMAGE}
+  local node_image=${KUBE_NODE_IMAGE}
+
+  ROOT_DEVICE_MASTER=$($AWS_CMD describe-images --image-ids ${master_image} --query 'Images[].RootDeviceName')
+  if [[ "${master_image}" == "${node_image}" ]]; then
```

Comment:
I think in general it would be fine to just make two calls here - bash is sufficiently messy that I'd rather trade-off scripting complexity for AWS efficiency of execution.  But thank you for doing the right thing - we should leave this as-is!